### PR TITLE
torch compile fast_newtonschulz

### PR DESCRIPTION
**Pull request by rishabh-ranjan**
made it almost 2x faster for my specific setting, but i expect it to give significant speedup in general too

### CODE REVIEW

**Comment by nil0x9** [comment]
Hi, what exactly is your setting and what device and problem shape did you test on?

**Comment by rishabh-ranjan** [comment]
I am training small transformer models (~25M params). GPU is A100. Do you not see any gains from torch.compile in your setup?

**Comment by rishabh-ranjan** [comment]
Exact times of optimizer.step are: 330 ms without torch.compile, 180 ms with torch.compile

**Comment by nil0x9** [comment]
> I am training small transformer models (~25M params). GPU is A100. Do you not see any gains from torch.compile in your setup?

For a small model like this, I think the dimensions of param matrices are very small too. In that case, I think a native matmul would suffice to work. In our benchmarking applying the method in this repo to small-dim matrices is actually slower than native matmul.

**Comment by rishabh-ranjan** [comment]
I see. Just FYI for my models, not using fast_newtonschulz is 230ms which is still slower than torch compiled fast_newtonschulz (180ms).
